src/host/posix/cli_daemon.cpp

```diff
@@ -47,67 +47,199 @@
-                    -1,
-                strerror(errno));
+    VerifyOrExit(
+        bind(mListenSocket, reinterpret_cast<const struct sockaddr *>(&sockname), sizeof(struct sockaddr_un)) != -1,
+        error = OTBR_ERROR_ERRNO);
+
+exit:
+    return error;
+}
+
+void CliDaemon::InitializeSessionSocket(void)
+{
+    int newSessionSocket = -1;
+    int flag             = -1;
+
+    VerifyOrExit((newSessionSocket = accept(mListenSocket, nullptr, nullptr)) != -1);
+
+    VerifyOrExit((flag = fcntl(newSessionSocket, F_GETFD, 0)) != -1, close(newSessionSocket));
+
+    flag |= FD_CLOEXEC;
+
+    VerifyOrExit((flag = fcntl(newSessionSocket, F_SETFD, flag)) != -1, close(newSessionSocket));
+
+#ifndef __linux__
+    // some platforms (macOS, Solaris) don't have MSG_NOSIGNAL
+    // SOME of those (macOS, but NOT Solaris) support SO_NOSIGPIPE
+    // if we have SO_NOSIGPIPE, then set it. Otherwise, we're going
+    // to simply ignore it.
+#if defined(SO_NOSIGPIPE)
+    VerifyOrExit((flag = setsockopt(newSessionSocket, SOL_SOCKET, SO_NOSIGPIPE, &flag, sizeof(flag))) != -1,
+                 close(newSessionSocket));
+#else
+#warning "no support for MSG_NOSIGNAL or SO_NOSIGPIPE"
+#endif
+#endif // __linux__
+
+    Clear();
+
+    mSessionSocket = newSessionSocket;
+    otbrLogInfo("Session socket is ready");
+exit:
+    if (flag == -1)
+    {
+        otbrLogWarning("Failed to initialize session socket: %s", strerror(errno));
+        Clear();
+    }
 }
 
-void CliDaemon::Init(const std::string &aNetIfName)
+otbrError CliDaemon::Init(const std::string &aNetIfName)
 {
+    otbrError error = OTBR_ERROR_NONE;
+
     // This allows implementing pseudo reset.
-    VerifyOrExit(mListenSocket == -1);
+    VerifyOrExit(mListenSocket == -1, error = OTBR_ERROR_INVALID_STATE);
 
-    mNetifName = aNetIfName;
-    CreateListenSocketOrDie();
+    SuccessOrExit(error = CreateListenSocket(aNetIfName));
 
     //
     // only accept 1 connection.
     //
-    VerifyOrDie(listen(mListenSocket, 1) != -1, strerror(errno));
+    VerifyOrExit(listen(mListenSocket, 1) != -1, error = OTBR_ERROR_ERRNO);
+
+exit:
+    return error;
+}
+
+void CliDaemon::Clear(void)
+{
+    if (mSessionSocket != -1)
+    {
+        close(mSessionSocket);
+        mSessionSocket = -1;
+    }
+}
+
+void CliDaemon::Deinit(void)
+{
+    Clear();
+}
+
+void CliDaemon::UpdateFdSet(MainloopContext &aContext)
+{
+    if (mListenSocket != -1)
+    {
+        aContext.AddFdToSet(mListenSocket, MainloopContext::kErrorFdSet | MainloopContext::kReadFdSet);
+    }
+
+    if (mSessionSocket != -1)
+    {
+        aContext.AddFdToSet(mSessionSocket, MainloopContext::kErrorFdSet | MainloopContext::kReadFdSet);
+    }
+
+    return;
+}
+
+void CliDaemon::Process(const MainloopContext &aContext)
+{
+    ssize_t received;
+
+    VerifyOrExit(mListenSocket != -1);
+
+    VerifyOrDie(!FD_ISSET(mListenSocket, &aContext.mErrorFdSet), strerror(errno));
```

Comment:
I saw `VerifyOrDie`s are replaced by `VeriyOrExit` at other places. I wonder why it's not replaced here?